### ISSUE TITLE
Instructions for ddev-drupal-solr and Drupal 7

### ISSUE THREAD

**Issue by jenlampton** [ddev/ddev-drupal9-solr#44]
I expect there will be a handful of larger Drupal 7 sites suddenly getting attention now that EOL draws near. Could we add some docs on how to get Drupal 7 + Search API running with this add-on? I have not figured it out yet, but if/once I do I'll be happy to file a PR against the README if interested.

**Comment by tyler36** [comment]
For context, Drupal 7 is EOL on 05 Jan 2025; around 3 weeks from today.

**Comment by jenlampton** [comment]
> Drupal 7 is EOL on 05 Jan 2025; around 3 weeks from today.

Yes, exactly!  The "lets wait and see" people (often those with larger / more complex D7 sites using solr) are out of time to wait, and are doing things with their old sites now :D In some cases using new developers (ahem) who need to get these old D7 sites working on ddev. 

I was able to get my site working on ddev using `ddev-drupal-solr` after failing with `ddev-solr`, so this project has proved very helpful ❤ 

My steps were as follows:

**For ddev:**

0) Install this add-on with `ddev add-on get ddev/ddev-drupal-solr && ddev restart`

1) Downgrade the Solr verison to at most verison 7 - see [supported solr versions for Search API](https://www.drupal.org/project/search_api_solr). 

Edit the `.ddev/docker-compose.solr.yaml` file, replace `image: solr:8` with `image: solr:7`

2) Add the corresponding schema (and other things as needed) for version 7.  I got these from [the D7 repo for the search_api_solr module](https://git.drupalcode.org/project/search_api_solr/-/blob/7.x-1.x/solr-conf/7.x/schema.xml?ref_type=heads). and add that into `.ddev/solr/config`. 

I did `cp sites/all/modules/contrib/search_api_solr/solr-conf/7.x/* .ddev/solr.conf/.` but this is based on the lcoation of your module and ddev config. 

3) Restart, and voila! Confirm Solr is working by visiting `http://mysite.ddev.site:8983/solr/`.

**In Drupal:**

1) Set up a new Search API server. The values I used are as follows: protocol: `http`, host: `solr`, port: `8983`, solr path: `/solr/dev` (where `dev` is the name of the solr "core").

2) Then set up a new Search API index, and index the site. 

3) Build a new View on the index, and then you have a search!

**Comment by tyler36** [comment]
PRs are always welcome. They may take some to get merged, especially at this time of year, but other developers will find the information helpful!

**Comment by rfay** [comment]
Yes @jenlampton please do a PR on the README adding a section "Installation on Drupal 7" with your info. Super easy. You can even use the Github web UI to do it (click the pencil at upper right on https://github.com/ddev/ddev-drupal-solr/blob/main/README.md )

**Comment by jenlampton** [comment]
I have created https://github.com/ddev/ddev-drupal-solr/pull/45/files, which is maybe too much information. I'm going to create an alternate PR that removes all the views instrucitons incase that would be preferable.

**Comment by jenlampton** [comment]
Okay, this one might be more palettable :) 
https://github.com/ddev/ddev-drupal-solr/pull/46/files

**Comment by rfay** [comment]
Just FYI, you don't have to create new PRs to change things, you can push new info into an existing PR.

**Comment by jenlampton** [comment]
@rfay but then how to do compare the two options? Oh -- would you prefer separate commits? I usually like to have all of my change in a single commit. I suppose they could be squashed easily enough! 

We usually do separate PRs to compare options. Force of habit, I suppose!

**Comment by rfay** [comment]
It's easier for reviewers if you just do the thing you think is best, then reply to suggestions. The bottom line is that you're the reigning expert here, and your suggestion is going to get accepted without trouble.

**Comment by jenlampton** [comment]
My preference is for [the more verbose documentation](https://github.com/ddev/ddev-drupal-solr/pull/45/files), since that information was hard to locate. I'll close the other PR and we can adjust as per feedback.
